### DISCUSSION TITLE
Name of Youtube download changes to the name of the video that was viewed before or just says Youtube -01

### DISCUSSION THREAD

**Discussion by mackfire** [aclap-dev/video-downloadhelper#1132] [Help and Q&A]
### Your question

When I download a YouTube video the right name of the video shows correctly during the download and when finished it changes to the name of the video viewed before this one or just names it Youtube-01, 02, etc.  Is there something wrong in the setting?  Mac OS 11.7.10, Firefox is up to date, Download  Helper 9.1.0.42 

### Addon & CoApp version and details

When I download a YouTube video the right name of the video shows correctly during the download and when finished it changes to the name of the video viewed before this one or just names it Youtube-01, 02, etc.  Is there something wrong in the setting?  Mac OS 11.7.10, Firefox is up to date, Download  Helper 9.1.0.42

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I am just another user.

You really should give all the information provided by the `Copy information to clipboard` button in the VDH settings.  Sometimes there are little clues in there that come in handy.

Also, let's have some URLs from YouTube that are giving you this effect.  I have not observed this so maybe it's something specific to the URLs you are visiting.  Let's see if I get the same results you do on the same videos.

**Reply by mackfire** [reply]
Its just any video downloaded from YouTube.  No exceptions that I can find.  Really there is none that keep their original title after the download is complete.  It always names the current download with the previous title from the one you downloaded before it.

**Reply by GrampaWildWilly** [reply]
That's not what's happening to me.  That's why I want you to give me a couple on which you observed this happening.  Also, post all your `Copy information to clipboard` stuff.

**Comment by boyjah** [comment]
I am having the exact same problem! It only started recently (maybe 2 months ago?)

version: 9.1.0.42
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"dea7b9f7ee754020"}
platform: aarch64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:129.0) Gecko/20100101 Firefox/129.0
{
    "valid_license_message_has_been_displayed": true,
    "download_directory": "/Users/randalirwin/Downloads",
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": true,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": true,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 8,
    "successfull_dl": 23
}

**Comment by GrampaWildWilly** [comment]
Why is it so hard for people to post some actual URLs of problematic videos?  Look, I'm having no problems whatsoever with YouTube.  Therefore, there is no problem on YouTube.  You are making it all up.  Disagree with me?  Then post some URLs!

**Reply by mackfire** [reply]
https://youtu.be/OFHBoVArFug?t=24
https://youtu.be/rn4XvpOtQm0?t=17
https://youtu.be/bcq2MHxbmWA?t=23

**Reply by GrampaWildWilly** [reply]
Finally!  Now I can work with this.

I am doing this with Windows 11 Pro 64-bit version 23H2 build 22631.3958, Firefox 129.0 64-bit, licensed VDH 9.1.0.41 beta, CoApp 2.0.20.

I started by visiting your first YouTube URL.  For some reason, VDH put up 2 possible downloads.  I have no idea why.

For any of these images, if it's too small to read, click it.  GitHub will open the image in a new browser window.  Then you can use the browser's own magnifier (Ctrl-+) to make things legible.

![#01](https://github.com/user-attachments/assets/f359d112-53ba-4e9b-b3ee-4285bff104f8)

I deleted the second one, then opened the list of variants on the remaining one.  Since there was a 2160p variant on offer, I decided to get that one.

![#02](https://github.com/user-attachments/assets/02097210-5a3b-4225-9db6-9657b5743b0a)

Contrary to my usual practice, I accepted the file name VDH proposed without altering it.

![#03](https://github.com/user-attachments/assets/d44012bb-7f92-445f-8dcf-dbc3400d7157)
![#04](https://github.com/user-attachments/assets/5408eebc-ae8e-46ab-8f11-6dda52998d2a)
![#05](https://github.com/user-attachments/assets/26c217d4-3660-498c-9010-603b9d1fe6a7)

Here's what I downloaded.  It played fine in VLC.  I didn't sit & watch it, just skimmed it.  There was video & audio at each of my sample points.

![#06](https://github.com/user-attachments/assets/1ac375ea-9378-4e44-9461-c24cd0ad2c91)

**Reply by GrampaWildWilly** [reply]
Then I copy/pasted your second URL into the URL bar of the YouTube window I already had open on your first URL.  I don't know what's the deal with VDH recognizing 2 videos.  I haven't seen that before.  Whatever.  I deleted the second one & went through the same process again.  You'll note that once again, I just accepted the file name proposed by VDH.  Once again, sampling the result file showed it was downloaded just fine.

![#07](https://github.com/user-attachments/assets/4de103bf-b117-4374-b2a8-531a8c5d52d5)
![#08](https://github.com/user-attachments/assets/04208c6f-a530-42d4-ba03-46f46487d668)
![#09](https://github.com/user-attachments/assets/40e611c6-e6af-44f7-9e11-ea18e10703eb)
![#10](https://github.com/user-attachments/assets/e58bbbce-30cb-4ea6-8235-3c5fb866f763)
![#11](https://github.com/user-attachments/assets/eb41e48d-99c8-4e2d-b2ad-ad2c744fa45c)
![#12](https://github.com/user-attachments/assets/641d31dd-5e3e-4688-8765-fa9c1c81b512)

**Reply by GrampaWildWilly** [reply]
And the same deal with your third URL.

![#13](https://github.com/user-attachments/assets/0b891b37-e534-430e-b2a2-1037d7cf87ce)
![#14](https://github.com/user-attachments/assets/7a0ab9f7-9e46-410f-8273-b3388b5b6a70)
![#15](https://github.com/user-attachments/assets/421cb124-0cec-47d2-b675-43c62e5a7125)
![#16](https://github.com/user-attachments/assets/33056fa8-c0e5-4c0a-ade0-94df10276524)
![#17](https://github.com/user-attachments/assets/60393697-7dd1-40be-8592-2aafcd18bd66)
![#18](https://github.com/user-attachments/assets/48393f9b-f061-4550-a672-99026a450659)

So I can't replicate your results.  The file naming doesn't just add a number on the end.  Also, with each successive download, it doesn't get stuck picking up the previous name.

Now, there is the difference in the operating system.  You guys are running Mac while I'm on Windows.  I have no idea how that might affect things.  I know less than diddly squat about Mac.  I've never used one.

Then there's probably the difference in our versions of VDH.  I'm running the beta.  I have been running VDH betas for at least the past 5 years.  Everybody should be running the beta.  For some products, the beta version might be an experimental version full of bugs.  With VDH, that is not the case.  The betas are interim versions that contain bug fixes.  Every so often, the latest beta gets promoted to the general availability version.  So if you run the beta, you get the bug fixes early.  That's worth doing.

**Comment by boyjah** [comment]
I always see 2 identical options for download as shown in your images. I always use the top one. If I download more than 1 file in a session, the file name problem happens. And as the original poster explained, the right name of the video shows correctly during the download and when finished it changes to the name of the video viewed before this one or just names it Youtube-01, 02, etc.

**Reply by GrampaWildWilly** [reply]
That is really strange.  Please post a series of screenshots like I did.  Maybe somebody will notice something & suggest an answer.  Or maybe it really is a bug in VDH on the Mac.  I'm sure @paulrouget Paul would want to see what you're doing in order to determine whether there's anything he could do about it.  So, screenshots from you would seem to be in order here.

**Comment by boyjah** [comment]
I am ready to download a file and I see:

<img width="344" alt="Screenshot 2024-08-13 at 00 14 04" src="https://github.com/user-attachments/assets/921f8bba-f2a1-4dd4-88fd-563b95b695b4">


After downloading, it shows as:

<img width="349" alt="Screenshot 2024-08-13 at 00 14 43" src="https://github.com/user-attachments/assets/f01d5520-f3af-4c3a-a851-f982eec76232">


I prepare to download the next file and it shows:

<img width="341" alt="Screenshot 2024-08-13 at 00 15 41" src="https://github.com/user-attachments/assets/0bab9483-0b97-4036-85a2-0ce4e5f12808">


But after downloading, this is what it shows:

<img width="343" alt="Screenshot 2024-08-13 at 00 16 00" src="https://github.com/user-attachments/assets/54dde2dc-a36a-421c-8523-df75a3b70c13">

As you can see, it gave the first file a random Youtube name, and then the second file was given the name of the first downloaded file

As I said before, this did not happen until a couple of months ago and I have been using Download Helper for years

**Reply by Mattmjs** [reply]
I see the same behavior too, the name of the first downloaded video comes up in the second video in the save as dialog.
Although I don't use it much on YouTube as I have something else for that.

**Reply by boyjah** [reply]
What do you use for YouTube? I want to try it!

**Reply by Mattmjs** [reply]
I don't know I should talk about it here as might be off topic, the other thing is it only has 360p,720p & 1080p.
Edit: emailed you.

**Reply by GrampaWildWilly** [reply]
That is really freaky.  I do see that you are using `Download` but I was using `Download As...` in my examples above.  So I switched my default action to `Download` & downloaded 2 of the same videos that I did above.  I'm not seeing the strange renaming after the fact.  Here's the 2 files I got:

![image](https://github.com/user-attachments/assets/d26922d9-db5d-4e86-9f4f-6411d893f547)

That's a cropped screenshot of the Windows file manager.  I got the file names anybody would expect & anybody would want.

I think we're looking at a bug that is unique to Mac.  Probably only Paul could determine where exactly the bug lies.  Maybe he just needs to use a different API for naming the files.  I'm guessing, of course.  I don't have access to the code.  I'm just another user, like you.

**Reply by GrampaWildWilly** [reply]
It occurs to me that Smartnaming might have an impact.

![#01](https://github.com/user-attachments/assets/fae9924e-b7ee-41f1-87a4-5be2c6933fba)
![#02](https://github.com/user-attachments/assets/c181a9a6-43df-47b0-b44d-fda285949fad)
![#03](https://github.com/user-attachments/assets/d9d2978c-61e6-4bee-9010-6b8300f28d1a)
![#04](https://github.com/user-attachments/assets/c5376461-c51a-44a2-8314-b823080ea020)
![#05](https://github.com/user-attachments/assets/4bb88362-1be5-41f7-bf26-428339fba92e)
![#06](https://github.com/user-attachments/assets/854c6cf1-6237-4531-8064-84e8836a9795)

**Reply by GrampaWildWilly** [reply]
![#07](https://github.com/user-attachments/assets/7315fba6-2a94-4a41-abf4-b65422d7f68e)
![#08](https://github.com/user-attachments/assets/c53d6393-db9e-475a-a5eb-a8db31248b3e)
![#09](https://github.com/user-attachments/assets/f16b307d-acc0-4bd6-83fc-0c0126ff6ef3)
![#10](https://github.com/user-attachments/assets/dd2cc6be-e758-49be-8337-405187447b8c)
![#11](https://github.com/user-attachments/assets/880980a4-5fb9-4e49-b9f8-f3d44540ddfa)
![#12](https://github.com/user-attachments/assets/866054a9-e712-470a-a7b1-91cf05f2a876)
![#13](https://github.com/user-attachments/assets/82bff1d4-1d86-4baf-b978-f052f5993ae8)

**Reply by GrampaWildWilly** [reply]
If you look closely you'll see that when I highlighted the video title, my highlight included an exclamation point at the end.  I can't tell if the gibberish string I pasted into the Smartnaming dialog purposely omits the exclamation point.  It's not entirely clear to me whether the omission of the exclamation point is something the Inspector did or something VDH did.  Either way, I think it's wrong.  Exclamation point is a valid character in file names.  But that's a minor quibble.  This Smartnaming rule will take whatever title YouTube has applied to a video & use it to name the file you download with VDH.  It's highly unlikely any given rule will work on multiple web sites.  The button for `Save for all hostnames` would be useful for applying the `Max length` value globally.

It may not be obvious but you do not have to go through this rigamarole to define a Smartnaming rule for every YouTube video.  This rule tells VDH to use the video's title no matter what video you are downloading from YouTube.  When you press the Test button, it shows you what it will do for the current video.  On another video, you can keep the same rule & Test will show whatever VDH will use for that video.  There's a reason there is no button in the Smartnaming dialog for "Apply to just this video."  That's just not how Smartnaming works.  A rule applies to an entire web site.  The assumption is that all the pages on a web site are coded more or less the same.  I would expect that only a pathological web site would resist VDH's Smartnaming algorithm.

I suggest Smartnaming because I have a hunch, a suspicion, a wild-ass guess that it might affect the pathological naming effect you are seeing.  I would very much like to know if I've guessed right.  I am totally prepared to hear that I'm wrong.  But I'd like you to try this.  Like I said before, I'm not seeing any problem like this on Windows.  So it would very likely be extremely helpful to Paul if you could try this.

**Reply by Mattmjs** [reply]
Did you see my comment, get the same problem as the others said. In the response by @boyjah  in the images they chose mp4.
I selected mp4 options for the download. The only difference I see is you went with the MKV options.
You should try an mp4 option without the smartnaming. The problem is with default name that is proposed.

**Reply by GrampaWildWilly** [reply]
![image](https://github.com/user-attachments/assets/f6d55989-d44c-4213-b621-92902decfe53)

I turned off Smartnaming & downloaded the MP4s from my test above.  Download, not Download As.  The proof that I turned off Smartnaming is that the file names have " - YouTube" tacked on the end, which has been the default behavior in VDH for years.  I hovered the mouse over the one with the long name so you could see the entire file name.

Once again, I'm not seeing this.  I don't know how I'm not seeing it, but I'm not.  I'm on Windows.  These other guys are on Mac.  That's why I thought it was a Mac problem.  But if you're seeing it on Windows, then I'm just baffled.  Paul would probably want you to do the trick with capturing a log.  That might be helpful to him.  Another user posted a log yesterday & I looked at it.  I couldn't make heads or tails of it.  I sure hope Paul finds it useful.

**Reply by GrampaWildWilly** [reply]
Meanwhile, since you @Mattmjs Matt are seeing this on Windows, maybe Smartnaming would correct it for you.  Try that.

**Reply by Mattmjs** [reply]
Ok after thinking about this more I think it is not the the formats but the stable extension having the problem. 
It isn't a problem in the beta version after remembering you have the beta, after all it's in your version information further up in the discussion and I have it in another profile. 

I didn't see the problem in the beta , the Stable version has the bug then.

**Reply by GrampaWildWilly** [reply]
In that case, I can only repeat the advice I already gave to the Mac guys: Get the beta.

Paul seems to have stopped posting betas for Chrome & Edge.  They're not in the Wiki & they used to be.  Are the betas still on the respective stores online?

**Reply by GrampaWildWilly** [reply]
A quick search of the 2 stores turns up VDH but not VDH beta.  Seems to me the betas used to be on the stores.  I know Paul is biased.  But he was doing betas for all 3 browsers for a while.  I suppose it is a lot of work.  Oh well.  I like Firefox.  I don't use either of the other 2 browsers.  I don't have Chrome & Edge comes up here only by accident, which is rare.

**Comment by mackfire** [comment]
Thanks for all of the info everyone.  I just tried the MKV 1080P download and it did the same thing as MP4.  Named it Youtube.mkv.  You all are above my pay grade and the in and outs of deep diving into code.  I am just a licensed used that wants it to work like it used to.  I really do not want to pay for something and not have it work, just like my customers are with me in my business.  I hope that a fix is in the works.

**Reply by GrampaWildWilly** [reply]
@Mattmjs Matt & I already explained what the fix is.  You need to install the VDH beta.  The beta includes the bug fix for the problem you are seeing.  It can't possibly be above your pay grade to install a version of VDH.  You've already done that.  So install the beta & your problem will be solved.

**Comment by boyjah** [comment]
I installed Version 9.1.0.41 (beta), target mozilla, locale en-US and the problem is not solved

**Reply by GrampaWildWilly** [reply]
Well then, maybe it is bug that is unique to Mac.  I don't know what to do next.  Like I said before, I don't know a thing about Mac.   I assume you've closed all browser windows at least once since you installed the VDH beta.  I assume the VDH settings do recognize the CoApp.  If that's all good, I don't know how to diagnose this.  I think we're going to need @paulrouget Paul's help now.

**Comment by boyjah** [comment]
Yes, I cleared browser cache then quit the app and restarted. And yes, the CoApp shows as installed and functioning...

<img width="321" alt="Screenshot 2024-08-13 at 22 30 31" src="https://github.com/user-attachments/assets/1d38a1f1-254c-4109-a5e3-59e0592f191c">

**Reply by Mattmjs** [reply]
That's surprising because it works as expected when I tested the beta, I tried two more examples and they each have their own
names.

**Comment by steveyyy8** [comment]
Hello
I have the same problem. **It started exactly after update to the latest version 9.1.0.42.** Since there has been no fix offered, I'm posting now.

version: 9.1.0.42
target: mozilla
channel: stable
lang: de
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"5b42b3684dee42c5"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:129.0) Gecko/20100101 Firefox/129.0
{
    "valid_license_message_has_been_displayed": true,
    "http_media_download_strategy": "coapp",
    "used_history_button": true,
    "download_directory": "C:\\Users\\steve\\Downloads",
    "concurrent_downloads_max": 5,
    "show_thumbnail_in_notification": false,
    "show_success_notification": false,
    "view_options": {
        "all_tabs": false,
        "low_quality": true,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": false,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": true
    },
    "show_context_menu": false,
    "forget_media_on_tab_close": false,
    "yt_warning": false,
    "use_wide_ui": false,
    "use_legacy_ui": false,
    "use_sidebar": false,
    "open_count_store": 34,
    "successfull_dl": 28,
    "never_show_successfull_dl_message": true,
    "history_limit_in_days": 2
}

My typically workflow is to open my pefered sites I want to check if new videos appear. 25+ Pages at once (for later offline view). Then close pages I don't like to download. For the other tabs, I push the download button once after another, page by page. The "Youtube[numer]" naming appears sporadically, somethimes more, somtimes less.
I did not make more investigations so far..

* The same Video can have the youtube name, but after page refresh and download again, it then may have the correct name of the video.
* I have the feeling that the more videos has been open before in tabs, the bigger the change in having the name issue.

Maybe a CPU Speed issue (too fast?), some page unload from RAM memory when many tabs are open?...

I have already tried the most important settings, without any change. Prefering download with CoApp or w/o CoApp is without change of behavour.
I never changes any settings regarding the intelligent naming, before until now, but also no changes in occurenace

I can create some log if I know how.

**Reply by GrampaWildWilly** [reply]
There's instructions for creating a log in the Wiki.  There's a link to the Wiki at the top of every page of this forum.  I've looked at some of those logs & they don't tell me anything.  But then, I'm just another user.  I wouldn't bother with the log unless @paulrouget Paul asks for it.  He's the developer.  I would assume a log would be helpful to him.

In the meantime, it seems that using the beta on Mac is perhaps not a solution.  But both Matt & I have been downloading from YouTube using the VDH beta on Windows & we're not seeing the problem.  Unlike the other users in this discussion, you are also using Windows.  I hold out hope the beta would cure this problem for you.  The beta is also in the Wiki.  It can't hurt to try.  Do come back & let us know how the beta works for you.

**Reply by steveyyy8** [reply]
Yesterday I had a complete name mismatch. One download file got a wrong name of another video. The name was correct but should have been assigned to another simultaneouse download. Other Video had again the Youtube-xx name.
I suspect some problem with multithreading or multicore. Maybe some atomic sections missing (if this is possible for firefox plugins..)
But problem also present when FF is set to use one CPU core only.

**Reply by steveyyy8** [reply]
Since a few days, I'm using the OLD GUI. The issue **did not occure** since then.

Can someone else try this?

(The new GUI is more friendly. But I's a workaroung in the meanwhile.)

**Comment by dee-max** [comment]
Same problem with me as well. Incorrect name of video, while in playlist it is detecting previous played video name as the current downloaded.

version: 9.1.0.42
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"b67472ff1a1e4adf"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:129.0) Gecko/20100101 Firefox/129.0
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 3,
    "successfull_dl": 6,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": []
    }
}

**Reply by Mattmjs** [reply]
Did you read the discussion first or did you just post your problem. There is a solution.

**Comment by boyjah** [comment]
None of the solutions on this thread have worked for me. The problem persists after having tried all suggestions for a fix.

**Reply by steveyyy8** [reply]
Can you try the old GUI Style? For me, the old gui works fine since I switched back.

**Comment by jarmake** [comment]
I am getting the same result with BETA
First download after opening browser is as it should be. Then the following one gets the name of the previous one. When the tab is open and download begins, the video name is initially detected correctly, but when the download finishes, the name gets overwritten with the name of the previous one. This started for me with stable vesion 9.1.0.42. 

version: 9.1.0.41
target: mozilla
channel: beta
lang: en-GB
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"d1aba1dbb5324760"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:129.0) Gecko/20100101 Firefox/129.0
{
    "valid_license_message_has_been_displayed": true,
    "download_directory": "I:\\ytu\\BL",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": true,
    "open_count_store": 3,
    "successfull_dl": 72,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": []
    }

**Reply by Mattmjs** [reply]
Using the beta extension, I don't have the problem. Maybe it's a regional / geographic issue, you could see if that makes
a difference using a vpn and changing to different countries.

![27-08-2024-4;42](https://github.com/user-attachments/assets/4be791a6-13c4-4270-9730-7dbc6303e1bb)

**Comment by kebera** [comment]
Same problem on Linux.
`version: 9.1.0.42
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"/home/keb/.local/share/vdhcoapp/vdhcoapp","version":"2.0.19","new_version":false}
license: {"unneeded":true}
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/115.0
{
    "download_directory": "/home/keb/Downloads/dwhelper",
    "theme": "system",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": false,
    "open_count_store": 2,
    "successfull_dl": 39
}`

**Comment by mister-bum** [comment]
As @kebera stated, the problem exists also on Linux (in my case kubuntu 22.04):

```
version: 9.1.0.44
target: mozilla
channel: stable
lang: de
coapp: {"found":true,"path":"/home/myuser/vdhcoapp-2.0.19/vdhcoapp","version":"2.0.19","new_version":false}
license: {"unneeded":true}
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:132.0) Gecko/20100101 Firefox/132.0
{
    "used_history_button": true,
    "view_options": {
        "all_tabs": true,
        "low_quality": false,
        "sort_by_status": false,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": true,
    "open_count_store": 6,
    "successfull_dl": 21,
    "never_show_successfull_dl_message": true,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": []
    }
}
```

- I have experienced this problem only on YouTube yet - other video hosting sites do not seem to be affected.
- Also the problem does not seem to be FireFox related, as it occurs in Vivaldi (Chrome) too.
- Last, it is not ONE CERTAIN video/url that is affected, but it happens at a rate of 99% of all videos, hence I do not post any URLs here, but describe the steps to reproduce the error.

**On YouTube:**
Providing that autoplay is enabled, after the video#1 has finished, video#2 starts and the entry in the VDH bar is "refreshed" to the title of video#1. After video#2 has finished, video#3 starts and the entry in the VDH bar is refreshed to the title of video#2 and so on.....

When you manually choose an entry in the playlist window, the behavior is the same, i.e. the title of the last played video is used. 
Interestingly, only the title is affected - the length of the video is refreshed correctly, so I wonder whether this a "transmission error"...might be some variable that is not properly updated and still contains the old title....just a feeling.

When you manually refresh the website, the video titles is recognized correctly, but this often leads to a double entry.

**Comment by GrampaWildWilly** [comment]
It occurs to me on rereading this thread that I always do `Download As`, never `Download`.  `Download As` causes VDH to open a file selection dialog BEFORE launching the download.  That lets me choose my own file name for the target file.  Maybe that's why I'm not seeing the problem.  I just do things differently from you folks.  Look at my images above that I posted a few months ago.  My VDH menu shows it's going to do a `Download As`.  You folks should consider changing your operating procedure to use `Download As` as your default action.  I would expect that would simply avoid the problem because you would be supplying your own file names.

Now, perhaps there is a bug in the `Download` function that makes it get confused when you're doing several `Download` functions in a row.  OK.  If you know that doesn't work, stop doing it.  Do something that will work.  Use `Download As` instead.  You don't really want to suffer until Paul comes up with a new version, do you?  So quit doing what fails & start doing what will work.

**Comment by mister-bum** [comment]
So first, @GrampaWildWilly, thanks for the suggested workaround which does not work for me, however. I just checked it out, and it doesn't matter whether I choose _direct download_ or _download as_ - in both cases the suggested name is the one of the previous video. One might argue about what is the lesser work, to rename it in the _download as_-dialog or later on your HD, but I won't go into that ;-)

Second, I  think the workflow might actually be crucial in order to reproduce the error: 

Open a YT window/tab and click on any of the links provided by YT and the error occurs as described in my initial post.
But if you copy that link (RMB, context menu....) and paste in the browser's address bar, the error does not occur. This is somewhat  "logical" to me, as a refresh of an already opened site also "cures" the problem. 

I have the feeling that when @mackfire provided those URLs above, you entered them one by one in the address bar.  If the error is consistent (and therefore also existing at your side) that procedure would not show the faulty behavior. 

To sum up my experiences/observations:

- the error occurs on YouTube only. Other sites don't seem to be affected.
- the error occurs when a site is loaded by clicking on one of the links provided on a YT-site
- the error does not occur when the URL is entered in the address bar
- the error does not seem to be related to a specific VDH function, as it doesn't matter whether one chooses _download as_ or _direct download_

**Reply by pkundrat** [reply]
I have the same issue (stable 9.3.0.1) - but i dont think it is the exactly the same issue.
OP reported that name in the list was originally detected correctly and only changed after download finished. My case is different, the name is wrong from the detection and stays unchanged until download is complete (i am using download as) - it is not clear how yours behaves, but looks similar to mine.
OP also reported that in the sequence, the current name always changed to the previous one. My case is different - the name stays the same in the subsequent videos from the playlist. Refresh fixes the issue (for the current video only), so the subsequent videos use the name from the last refresh. 
I tried playlist on YT and it works for me, it only fails on a paywall site.

Could people who have the issue please confirm if Refresh fixes the issue?
I believe we have two issues
1) from OP - download completion changes the name to the previous download
2) detected name in a playlist is from the first page (or the one that was refreshed last), download does not affect things

I will try beta to see if this works there.

**Reply by pkundrat** [reply]
I did one test - in Developer tools i disabled the event listener and the issue got fixed. So it is being caused by something event listener does that prevents VDH to refresh data for the new page. I will try to distil the faulty piece of page into static HTML to reproduce reliably and submit it for analysis (although probably that should belong to a separate topic as this looks like a separate issue)

**Comment by boyjah** [comment]
My workaround to this endless problem is to click the darker garbage can in the lower right corner of the menu that says:

Remove discovered and downloaded media
(files are not deleted)

Then I refresh the page and download.

Since doing that I have not had to rename any downloaded files...

**Reply by mister-bum** [reply]
Thanks for the suggestion, but I cannot see how this is different to just hitting the refresh button. This might be different from OS to OS or from browser to browser, but when I (kubuntu, vivaldi and firefox) refresh the page,  the VDH cue of the active tab is cleared automatically.

**Comment by steveyyy8** [comment]
The recent few month I was using the previous BETA Version with the old GUI as a workaround. Now the latest Version 9.3.0.1 which does not have the old gui, does not have the problem anymore for me. Seem to be solved (at least for me). The videonames does not mismatch anymore **[OK]**.

Another anomaly is not in my focus: When having many YT Tabs, and open each new Youtube link "open in new tab", the videoname detection is always **[OK]**.
**[not OK]** When having many YT tabs and click on a new video for open in "the current tab", then the videoname detection may file (stays at "youtube" name), when changing the tab to another tab, before loading has been finalized.

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user.

Legacy UI is not available in VDH 9.3.0.1 general availability for Firefox?

![#01](https://github.com/user-attachments/assets/0468321f-cb17-403c-a113-e6c10e4632b3)

I beg to differ.
